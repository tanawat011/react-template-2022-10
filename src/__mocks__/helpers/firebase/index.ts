import type * as Firebase from 'helpers/firebase'

const firebase = jest.createMockFromModule<typeof Firebase>('helpers/firebase')

module.exports = firebase
