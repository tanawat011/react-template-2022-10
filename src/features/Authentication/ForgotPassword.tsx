import { PATH } from 'Routes'
import { Button } from 'components/Button'

export const ForgotPassword = () => {
  return (
    <>
      <div id='forgot-password-form'>
        <h1>Forgot Password</h1>

        <form className='flex flex-col'>
          <label id='username'>Email</label>
          <input id='username' />

          <div className='flex'>
            <Button link to={`../${PATH.AUTH.LOGIN}`}>
              Back to Login
            </Button>
          </div>
        </form>
      </div>
    </>
  )
}
