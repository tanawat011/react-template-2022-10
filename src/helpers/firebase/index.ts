export * from './deleteCollection'
export * from './deleteDocument'
export * from './getAllDocuments'
export * from './getDocument'
export * from './getDocumentWithRef'
export * from './setDocument'
export * from './subscribeCollection'
export * from './subscribeDocument'
export * from './updateDocument'
