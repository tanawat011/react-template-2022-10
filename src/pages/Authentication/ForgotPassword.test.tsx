import { renderWithProviders } from 'helpers/test'

import { ForgotPassword } from './ForgotPassword'

describe('<ForgotPassword />', () => {
  test('match snapshot', () => {
    const { asFragment } = renderWithProviders(<ForgotPassword />)
    expect(asFragment()).toMatchSnapshot()
  })
})
