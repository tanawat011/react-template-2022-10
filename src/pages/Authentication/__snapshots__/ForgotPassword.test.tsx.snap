// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPassword /> match snapshot 1`] = `
<DocumentFragment>
  <div
    id="forgot-password-form"
  >
    <h1>
      Forgot Password
    </h1>
    <form
      class="flex flex-col"
    >
      <label
        id="username"
      >
        Email
      </label>
      <input
        id="username"
      />
      <div
        class="flex"
      >
        <button
          data-testid="button"
          to="../login"
        >
          Back to Login
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
